tdrs-backend/tdpservice/users/permissions.py

```diff
@@ -5,9 +5,12 @@
 
 def is_own_stt(request, view):
     """Verify user belongs to requested STT."""
-    return is_in_group(request.user, "Data Prepper") and (
-        request.user.stt.id == request.data['stt']
-    )
+    if (requested_stt := request.data.get('stt')) is None:
+        return True
+    else:
+        return is_in_group(request.user, "Data Prepper") and (
+            request.user.stt.id == requested_stt
+        )
```

Comment:
what is this code change about? If important, does this need to be tested, too?

Reply:
I initially tried to reuse this permission as it had been written for 431, and ran into issues because this endpoint does not provide an stt, so I had to gaurd against `request.data['stt']` being none by `get`ting it.

The assumption being, if the stt is not provided in any case it might be, it is known we are accessing the users own stt.

For this perticular endpoint, which stt is not important, but overall the permission structure is the same for this as for creating the meta data. only an OFA admin or a data prepper should be doing this.